### PR TITLE
added thirdweb docs and references

### PR DESCRIPTION
**Pull request by saminacodes**

### CODE REVIEW

File: docs/develop/thirdweb.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+:::
+
+# Overview
+
+thirdweb is a complete web3 development framework that provides everything you need to connect your apps and games to decentralized networks.
+
+## Create a contract using thirdweb
+
+To create a new smart contract using thirdweb CLI, follow these steps:
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:28]: Would add a quick intro for the CLI and would also link to the docs; something like:

```
[thirdweb CLI](https://portal.thirdweb.com/cli) is a command-line tool that provides developers with an interface to create, deploy, and manage smart contracts and dApps using thirdweb.
```

File: docs/develop/thirdweb.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+1. In your CLI run the following command:
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:30]:
```suggestion
1. In your CLI, run the following command to create a contract:
```

File: docs/develop/thirdweb.md (hunk 1 above)
Comment (DannyS03) [inline docs/develop/thirdweb.md:28]:
```suggestion
To create a new smart contract using thirdweb CLI, follow the step below.
```

File: docs/develop/thirdweb.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+   ```
+   npx thirdweb create contract
+   ```
+
+2. Input your preferences for the command line prompts:
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:36]:
```suggestion
2. Enter your preferred values for the command line prompts:
```

File: docs/develop/thirdweb.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+   1. Give your project a name
+   2. Choose your preferred framework: Hardhat or Foundry
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:38]:
```suggestion
   ii. Choose your preferred framework: [Hardhat](https://hardhat.org/) or [Foundry](https://getfoundry.sh/).
```

File: docs/develop/thirdweb.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+To create a new smart contract using thirdweb CLI, follow these steps:
+
+1. In your CLI run the following command:
+
+   ```
+   npx thirdweb create contract
+   ```
+
+2. Input your preferences for the command line prompts:
+   1. Give your project a name
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:37]:
```suggestion
   i. Give your project a name.
```

File: docs/develop/thirdweb.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+   2. Choose your preferred framework: Hardhat or Foundry
+   3. Name your smart contract
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:39]:
```suggestion
   iii. Name your smart contract.
```

File: docs/develop/thirdweb.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+   4. Choose the type of base contract: Empty, [ERC20](https://portal.thirdweb.com/solidity/base-contracts/erc20base), [ERC721](https://portal.thirdweb.com/solidity/base-contracts/erc721base), or [ERC1155](https://portal.thirdweb.com/solidity/base-contracts/erc1155base)
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:40]:
```suggestion
   iv. Choose the type of base contract: Empty, [ERC20](https://portal.thirdweb.com/solidity/base-contracts/erc20base), [ERC721](https://portal.thirdweb.com/solidity/base-contracts/erc721base), or [ERC1155](https://portal.thirdweb.com/solidity/base-contracts/erc1155base).
```

File: docs/develop/thirdweb.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+   5. Add any desired [extensions](https://portal.thirdweb.com/solidity/extensions)
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:41]:
```suggestion
   v. Add any desired [extensions](https://portal.thirdweb.com/solidity/extensions).
```

File: docs/develop/thirdweb.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+3. Once created, navigate to your project’s directory and open in your preferred code editor.
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:42]:
```suggestion
3. Once created, navigate to your project’s root directory and open your project in your preferred code editor.
```

File: docs/develop/thirdweb.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+4. If you open the `contracts` folder, you will find your smart contract; this is your smart contract written in Solidity.
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:43]:
```suggestion
4. If you open the `contracts` folder, you will find your smart contract; the contract you see is your smart contract written in Solidity.
```

File: docs/develop/thirdweb.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+
+   The following is code for an ERC721Base contract without specified extensions. It implements all of the logic inside the `[ERC721Base.sol](https://github.com/thirdweb-dev/contracts/blob/main/contracts/base/ERC721Base.sol)` contract; which implements the `[ERC721A](https://github.com/thirdweb-dev/contracts/blob/main/contracts/eip/ERC721A.sol)` standard.
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:45]:
```suggestion
   The following code is for an ERC721Base contract without specified extensions. It implements all of the logic inside the `[ERC721Base.sol](https://github.com/thirdweb-dev/contracts/blob/main/contracts/base/ERC721Base.sol)` contract, which implements the `[ERC721A](https://github.com/thirdweb-dev/contracts/blob/main/contracts/eip/ERC721A.sol)` standard.
```

File: docs/develop/thirdweb.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+           string memory _symbol,
+           address _royaltyRecipient,
+           uint128 _royaltyBps
+       ) ERC721Base(_name, _symbol, _royaltyRecipient, _royaltyBps) {}
+   }
+   ```
+
+   This contract inherits the functionality of ERC721Base through the following steps:
+
+   - Importing the ERC721Base contract
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:65]:
```suggestion
   - Imports the ERC721Base contract.
```

File: docs/develop/thirdweb.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+   - Inheriting the contract by declaring that our contract is an ERC721Base contract
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:66]:
```suggestion
   - Inherits the contract by declaring that our contract is an ERC721Base contract.
```

File: docs/develop/thirdweb.md [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+   - Implementing any required methods, such as the constructor.
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:67]:
```suggestion
   - Implements the required methods, including the constructor.
```

File: docs/develop/thirdweb.md [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+
+5. After modifying your contract with your desired custom logic, you may deploy it to [Polygon](https://thirdweb.com/polygon) using [Deploy](https://portal.thirdweb.com/deploy).
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:69]:
```suggestion
5. After modifying your contract with your desired custom logic, you may deploy it to [Polygon PoS mainnet](https://thirdweb.com/polygon) using [Deploy](https://portal.thirdweb.com/deploy).
```

File: docs/develop/thirdweb.md [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+Alternatively, you can deploy a prebuilt contract for NFTs, tokens, or marketplace directly from the thirdweb Explore page:
+
+1. Navigate to the [thirdweb Explore page:](https://thirdweb.com/explore)
+
+   ![img](/img/thirdweb/tw-explore.png)
+
+2. Choose the type of contract you want to deploy from the available options: NFTs, tokens, marketplace, and more.
+3. Follow the on-screen prompts to configure and deploy your contract.
+
+> For more information on different contracts available on Explore, check out [thirdweb’s documentation.](https://portal.thirdweb.com/pre-built-contracts)
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:82]:
```suggestion
> For more information on different contracts available on Explore, check out [thirdweb’s documentation](https://portal.thirdweb.com/pre-built-contracts).
```

File: docs/develop/thirdweb.md [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+
+   - Importing the ERC721Base contract
+   - Inheriting the contract by declaring that our contract is an ERC721Base contract
+   - Implementing any required methods, such as the constructor.
+
+5. After modifying your contract with your desired custom logic, you may deploy it to [Polygon](https://thirdweb.com/polygon) using [Deploy](https://portal.thirdweb.com/deploy).
+
+---
+
+Alternatively, you can deploy a prebuilt contract for NFTs, tokens, or marketplace directly from the thirdweb Explore page:
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:73]:
```suggestion
Alternatively, you can deploy a prebuilt contract an NFTs, token, or marketplace directly from the thirdweb Explore page. To do so, follow the steps below.
```

File: docs/develop/thirdweb.md [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+
+1. Navigate to the [thirdweb Explore page:](https://thirdweb.com/explore)
+
+   ![img](/img/thirdweb/tw-explore.png)
+
+2. Choose the type of contract you want to deploy from the available options: NFTs, tokens, marketplace, and more.
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:79]:
```suggestion
2. Choose the type of contract you want to deploy from the available options.
```

File: docs/develop/thirdweb.md [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+3. Follow the on-screen prompts to configure and deploy your contract.
+
+> For more information on different contracts available on Explore, check out [thirdweb’s documentation.](https://portal.thirdweb.com/pre-built-contracts)
+
+## Deploy a contract using thirdweb
+
+Deploy allows you to deploy a smart contract to any EVM compatible network without configuring RPC URLs, exposing your private keys, writing scripts, and other additional setup such as verifying your contract.
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:86]:
```suggestion
Deploy allows you to deploy a smart contract to any EVM-compatible network without configuring RPC URLs, exposing your private keys, writing scripts, and other additional setup such as verifying your contract.
```

File: docs/develop/thirdweb.md [hunk 20]
```diff
@@ -0,0 +1,118 @@
...
+
+1. To deploy your smart contract using deploy, navigate to the root directory of your project and execute the following command:
+
+   ```bash
+   npx thirdweb deploy
+   ```
+
+   Executing this command will trigger the following actions:
+
+   - Compiling all the contracts in the current directory.
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:96]:
```suggestion
   - Compiles all the contracts in the current directory.
```

File: docs/develop/thirdweb.md [hunk 21]
```diff
@@ -0,0 +1,118 @@
...
+   - Providing the option to select which contract(s) you wish to deploy.
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:97]:
```suggestion
   - Provides the option to select which contract(s) you wish to deploy.
```

File: docs/develop/thirdweb.md [hunk 22]
```diff
@@ -0,0 +1,118 @@
...
+   - Uploading your contract source code (ABI) to IPFS.
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:98]:
```suggestion
   - Uploads your compiled smart contract code (in the form of an Application Binary Interface, or ABI) to IPFS.
```

File: docs/develop/thirdweb.md [hunk 23]
```diff
@@ -0,0 +1,118 @@
...
+
+2. When it is completed, it will open a dashboard interface to finish filling out the parameters.
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:100]:
```suggestion
2. When deployment is complete, a dashboard interface will open to fill out the remaining parameters.
```

File: docs/develop/thirdweb.md [hunk 24]
```diff
@@ -0,0 +1,118 @@
...
+   - `_name`: contract name
+   - `_symbol`: symbol or "ticker"
+   - `_royaltyRecipient`: wallet address to receive royalties from secondary sales
+   - `_royaltyBps`: basis points (bps) that will be given to the royalty recipient for each secondary sale, e.g. 500 = 5%
+3. Select [Polygon](https://thirdweb.com/polygon) as the network
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:105]:
```suggestion
3. Select [Polygon](https://thirdweb.com/polygon) as the network.
```

File: docs/develop/thirdweb.md [hunk 25]
```diff
@@ -0,0 +1,118 @@
...
+4. Manage additional settings on your contract’s dashboard as needed such as uploading NFTs, configuring permissions, and more.
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:106]:
```suggestion
4. Once your smart contract is deployed, you can manage additional settings and functionalities through its dashboard. For example, you can upload NFTs, configure permissions and access control, add new features.
```

File: docs/develop/thirdweb.md [hunk 26]
```diff
@@ -0,0 +1,118 @@
...
+
+For additional information on Deploy, please reference [thirdweb’s documentation](https://portal.thirdweb.com/deploy).
+
+## Verify the Contract
+
+Contracts deployed to [Polygon](https://thirdweb.com/polygon) are automatically verified by thirdweb. You can check the status on the Sources view in the thirdweb dashboard.
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:112]:
```suggestion
Contracts deployed to [Polygon](https://thirdweb.com/polygon) are automatically verified by thirdweb. You can check the status on the "Sources" view in the thirdweb dashboard.
```

File: docs/develop/thirdweb.md [hunk 27]
```diff
@@ -0,0 +1,118 @@
...
+
+---
+
+### Need help?
+
+If you have any further questions or encounter any issues during the process, please [reach out to thirdweb support](https://support.thirdweb.com).
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:118]:
```suggestion
If you have any questions or encounter any issues during the process, **please reach out to the [official thirdweb support](https://support.thirdweb.com)**.
```

File: docs/develop/web3-development.md [hunk 28]
```diff
@@ -14,36 +14,29 @@ image: https://wiki.polygon.technology/img/polygon-logo.png
...
 
 ## Learn Web3 Programs
+
 - [Alchemy University (free)](https://university.alchemy.com)
 - [ConsenSys Academy Developer Program On-Demand course](https://consensys.net/academy/ondemand/)
 - [Dapp University](https://www.youtube.com/channel/UCY0xL8V6NzzFcwzHCgB8orQ)
 
 ## Smart Contract & Token Tutorials
+
+- [Deploy a Smart Contract on Polygon](https://blog.thirdweb.com/guides/deploy-a-smart-contract-on-polygon-matic/)
```
Comment (DannyS03) [inline docs/develop/web3-development.md:24]:
```suggestion
- [Deploy a Smart Contract on Polygon using thirdweb](https://blog.thirdweb.com/guides/deploy-a-smart-contract-on-polygon-matic/)
```

File: docs/develop/thirdweb.md [hunk 29]
```diff
@@ -0,0 +1,118 @@
+---
+id: thirdweb
+title: Deploy a Smart Contract Using thirdweb
+sidebar_label: Using thirdweb
+description: Guide to deploy smart contracts using thirdweb
+keywords:
+  - docs
+  - matic
+  - polygon
+  - alchemy
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:10]:
```suggestion
  - thirdweb
```

File: docs/develop/thirdweb.md [hunk 30]
```diff
@@ -0,0 +1,118 @@
+---
+id: thirdweb
+title: Deploy a Smart Contract Using thirdweb
+sidebar_label: Using thirdweb
+description: Guide to deploy smart contracts using thirdweb
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:5]:
```suggestion
description: "Lean how to deploy smart contracts on Polygon PoS mainnet using thirdweb."
```

File: docs/develop/thirdweb.md [hunk 31]
```diff
@@ -0,0 +1,118 @@
...
+   2. Choose your preferred framework: Hardhat or Foundry
+   3. Name your smart contract
+   4. Choose the type of base contract: Empty, [ERC20](https://portal.thirdweb.com/solidity/base-contracts/erc20base), [ERC721](https://portal.thirdweb.com/solidity/base-contracts/erc721base), or [ERC1155](https://portal.thirdweb.com/solidity/base-contracts/erc1155base)
+   5. Add any desired [extensions](https://portal.thirdweb.com/solidity/extensions)
+3. Once created, navigate to your project’s directory and open in your preferred code editor.
+4. If you open the `contracts` folder, you will find your smart contract; this is your smart contract written in Solidity.
+
+   The following is code for an ERC721Base contract without specified extensions. It implements all of the logic inside the `[ERC721Base.sol](https://github.com/thirdweb-dev/contracts/blob/main/contracts/base/ERC721Base.sol)` contract; which implements the `[ERC721A](https://github.com/thirdweb-dev/contracts/blob/main/contracts/eip/ERC721A.sol)` standard.
+
+   ```bash
```
Comment (DannyS03) [inline docs/develop/thirdweb.md:47]: If you're able to provide an example, to coincide with the template(s) and steps, that'd be great. If there is an example already available (for deploying on Polygon), we can also just link out to that.

**Review by DannyS03 — CHANGES_REQUESTED** [review]
Thank you for the contribution @saminacodes!

Added suggestions. Majority of them are nits

**Comment by saminacodes** [comment]
Hey @DannyS03 - made the changed. Let me know if there is anything else pending. Appreciate the help on this

**Review by DannyS03 — APPROVED** [review]
hey @saminacodes, thanks again! lgtm

**Pull request merged**